[Curl] Remove unused functions from CurlSSLHandle

kshukuwa opened the pull request:
#### 2f3f2e9a8c02877a11bbeec4f49353893bb41fd0
<pre>
[Curl] Remove unused functions from CurlSSLHandle
<a href="https://bugs.webkit.org/show_bug.cgi?id=257706">https://bugs.webkit.org/show_bug.cgi?id=257706</a>

Reviewed by Fujii Hironori.

Some functions in CurlSSLHandle and CurlContext are no longer needed.

* Source/WebCore/platform/network/curl/CurlContext.cpp:
(WebCore::CurlHandle::enableSSL):
(WebCore::CurlHandle::setUrl):
(WebCore::CurlHandle::enableSSLForHost): Deleted.
(WebCore::CurlHandle::setSslCert): Deleted.
(WebCore::CurlHandle::setSslCertType): Deleted.
(WebCore::CurlHandle::setSslKeyPassword): Deleted.
* Source/WebCore/platform/network/curl/CurlContext.h:
* Source/WebCore/platform/network/curl/CurlSSLHandle.cpp:
(WebCore::CurlSSLHandle::allowAnyHTTPSCertificatesForHost): Deleted.
(WebCore::CurlSSLHandle::canIgnoreAnyHTTPSCertificatesForHost const): Deleted.
(WebCore::CurlSSLHandle::setClientCertificateInfo): Deleted.
(WebCore::CurlSSLHandle::getSSLClientCertificate const): Deleted.
* Source/WebCore/platform/network/curl/CurlSSLHandle.h:

Canonical link: <a href="https://commits.webkit.org/264883@main">https://commits.webkit.org/264883@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bfbf2a493c5c6754ea5e712fe1f0188f30547899

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8926 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9214 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9432 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10579 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8906 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/8934 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11201 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9181 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11754 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9072 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10066 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/7865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/10738 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7361 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8165 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/15649 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/8470 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8313 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11637 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/8803 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7182 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8058 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12270 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1042 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8550 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fujii reviewed (commented):
Can we still accept a bad cert for a host?

kshukuwa committed:
[Curl] Remove unused functions from CurlSSLHandle
https://bugs.webkit.org/show_bug.cgi?id=257706

Reviewed by Fujii Hironori.

Some functions in CurlSSLHandle and CurlContext are no longer needed.

* Source/WebCore/platform/network/curl/CurlContext.cpp:
(WebCore::CurlHandle::enableSSL):
(WebCore::CurlHandle::setUrl):
(WebCore::CurlHandle::enableSSLForHost): Deleted.
(WebCore::CurlHandle::setSslCert): Deleted.
(WebCore::CurlHandle::setSslCertType): Deleted.
(WebCore::CurlHandle::setSslKeyPassword): Deleted.
* Source/WebCore/platform/network/curl/CurlContext.h:
* Source/WebCore/platform/network/curl/CurlSSLHandle.cpp:
(WebCore::CurlSSLHandle::allowAnyHTTPSCertificatesForHost): Deleted.
(WebCore::CurlSSLHandle::canIgnoreAnyHTTPSCertificatesForHost const): Deleted.
(WebCore::CurlSSLHandle::setClientCertificateInfo): Deleted.
(WebCore::CurlSSLHandle::getSSLClientCertificate const): Deleted.
* Source/WebCore/platform/network/curl/CurlSSLHandle.h:

Canonical link: https://commits.webkit.org/264883@main
